map/bookmark_manager.hpp

```diff
@@ -277,6 +277,7 @@ class BookmarkManager final
   kml::MarkGroupId GetCategoryId(std::string const & name) const;
 
   kml::GroupIdCollection const & GetBmGroupsIdList() const { return m_bmGroupsIdList; }
```

Comment:
There are cases where an unsorted list is ok, right? What are these cases?
```suggestion
  kml::GroupIdCollection const & GetUnsortedBmGroupsIdList() const { return m_bmGroupsIdList; }
```

Reply:
This case is ok when we want to check the count of groups. For example in the BookmarksListInteractor:
```swift 
  func canDeleteGroup() -> Bool {
    bookmarksManager.userCategories().count > 1
  }
```

This property is accessed a few times: 
when the sorted group list should be displayed 
and to check is the array is empty:
<img width="500" alt="image" src="https://github.com/organicmaps/organicmaps/assets/79797627/aca36748-9a22-4970-9eed-2df44214e1f9">

Reply:
This is not reliable compared to `bookmarksManager.userCategoriesCount()`